examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
+Author: [Hazem Essam](https://twitter.com/hazemessamm) and [Santiago L. Valdarrama](https://twitter.com/svpino)
+Date created: 2021/03/13
+Last modified: 2021/03/22
+Description: Siamese network with tf.data and custom training loop.
+"""
+
+
+"""
+TODO: Still need to work on this.
+
+# Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network) is used to solve
+many problems like detecting question duplicates, face recognition by comparing the
+similarity of the inputs by comparing their feature vectors.
+First we need to have a dataset that contains 3 Images, 2 are similar and 1 is different,
+they are called Anchor image, Positive Image and Negative image respectively, we need to
+tell the network that the anchor image and the positive image are similar, we also need
+to tell it that the anchor image and the negative image are NOT similar, we can do that
+by the Triplet Loss Function.
+Triplet Loss function:
+L(Anchor, Positive, Negative) = max((distance(f(Anchor), f(Positive)) -
+distance(f(Anchor), f(Negative)))**2, 0.0)
+Note that the weights are shared which mean that we are only using one model for
+prediction and training
+You can find the dataset here:
+https://drive.google.com/drive/folders/1qQJHA5m-vLMAkBfWEWgGW9n61gC_orHl
+Also more info found here: https://sites.google.com/view/totally-looks-like-dataset
+Image from:
+https://towardsdatascience.com/a-friendly-introduction-to-siamese-networks-85ab17522942
+![1_0E9104t29iMBmtvq7G1G6Q.png](attachment:1_0E9104t29iMBmtvq7G1G6Q.png)
+"""
+
+"""
+# Setup
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import random
+import tensorflow as tf
+
+from pathlib import Path
+from tensorflow.keras import applications
+from tensorflow.keras import layers
+from tensorflow.keras import losses, optimizers
+from tensorflow.keras import metrics
+from tensorflow.keras import Model
+from tensorflow.keras.applications.resnet import preprocess_input
+
+
+target_shape = (200, 200)
+
+
+"""
+# Load the dataset
+
+We are going to use the [Totally Looks Like dataset](https://drive.google.com/drive/folders/1qQJHA5m-vLMAkBfWEWgGW9n61gC_orHl). We are going to download it and unzip it inside the `~/.keras` directory.
```

Comment:
Suggestion:


```suggestion
We are going to load the Totally Looks Like dataset and unzip it inside the `~/.keras` directory
in the local environment.
```

I'd recommend to check the direct link - the original one is from Google Drive, not an external link. Perhaps we can use https://github.com/rosenfeldamir/TotallyLooksLike ? cc @fchollet